src/kfactory/kcell.py

```diff
@@ -1664,48 +1757,9 @@ def __copy__(self) -> KCell:
         """Enables use of `copy.copy` and `copy.deep_copy`."""
         return self.dup()
 
-    def add_port(
-        self, port: Port, name: str | None = None, keep_mirror: bool = False
-    ) -> Port:
-        """Add an existing port. E.g. from an instance to propagate the port.
-
-        Args:
-            port: The port to add.
-            name: Overwrite the name of the port
-            keep_mirror: Keep the mirror part of the transformation of a port if
-                `True`, else set the mirror flag to `False`.
-        """
-        if self._locked:
-            raise LockedError(self)
-        return self.ports.add_port(port=port, name=name, keep_mirror=keep_mirror)
-
-    def add_ports(
-        self,
-        ports: Iterable[Port],
-        prefix: str = "",
-        suffix: str = "",
-        keep_mirror: bool = False,
-    ) -> None:
-        """Add a sequence of ports to the cells.
-
-        Can be useful to add all ports of a instance for example.
-
-        Args:
-            ports: list/tuple (anything iterable) of ports.
-            prefix: string to add in front of all the port names
-            suffix: string to add the end of all the port names.
-            keep_mirror: Keep the mirror part of the transformation of a port if
-                `True`, else set the mirror flag to `False`.
-        """
-        if self._locked:
-            raise LockedError(self)
-        self.ports.add_ports(
-            ports=ports, prefix=prefix, suffix=suffix, keep_mirror=keep_mirror
-        )
-
     @classmethod
     def from_yaml(
-        cls: Callable[..., KCell],
+        cls,
```

Comment:
why remove the type hint here?

Reply:
Just seemed weird to type hint cls in this instance, i also added return type of Self, i think this is correct here

Reply:
fair enough

The problem with self-type is it's only >3.11

As google colab is dragging their feet (3 years for now I think), we need compatibility with 3.10 for now. But I believe you can import it from `typing_extensions` to make it work

Reply:
Can change to import from typing extensions, I have a habit of forgetting this and importing from typing always

Reply:
Already imported from typing extensions